src/views/pages/CreatorPage.vue

```diff
@@ -207,177 +207,166 @@
-      this.res.forEach((element) => {
-        let temp_packet = Object.keys(element);
-        Object.keys(element[temp_packet]).forEach((header) => {
-          if (/^-?\d+$/.test(element[temp_packet][header])) {
-            element[temp_packet][header] = Number(element[temp_packet][header]);
-          }
-          if (
-            header === "options" &&
-            element[temp_packet][header].includes(",")
-          ) {
-            element[temp_packet][header] = element[temp_packet][header]
-              .trim()
-              .split(",");
-          }
-        });
-      });
-      let newPacket = "";
-      try {
-        newPacket = await ApiService.createPacket(this.res);
-      } catch {
-        this.errorSw = !this.errorSw;
-        return;
-      }
-      this.createdHex = newPacket["hex"];
-    },
-    handleAdd(el) {
-      const protocol = el.srcElement.innerText;
-      if (
-        !this.pickedPackets.includes(protocol) &&
-        this.pickedPackets.length < 4
-      ) {
-        this.pickedPackets.push(protocol);
-      } else if (this.pickedPackets.includes(protocol)) {
-        this.pickedPackets.splice(this.pickedPackets.indexOf(protocol), 1);
-      }
-    },
-    changeCasing() {
-      if (this.upperCase) {
-        this.createdHex = this.createdHex.toLowerCase();
-        this.upperCase = false;
-      } else {
-        this.createdHex = this.createdHex.toUpperCase();
-        this.upperCase = true;
-      }
-    },
-    changeSpacing() {
-      this.createdHex = this.createdHex.replace(/\s+/g, "");
-      if (this.spacing) {
-        this.spacing = false;
-      } else {
-        for (let i = 0; i < this.createdHex.length; i += 3) {
-          this.createdHex =
-            this.createdHex.slice(0, i) + " " + this.createdHex.slice(i);
+import ApiService from "@/services/apiService";
+
+interface Packet {
+  name: string;
+  scapy_name: string;
+}
+
+@Component
+export default class CreatorPage extends Vue {
+  // Some protocols have different function name
+  // and "to_list" key (ex. Ethernet) so they
+  // have to be handled in core to be able to add them here
+  packets: Array<Packet> = [
+    { name: "Ether", scapy_name: "Ethernet" },
+    { name: "IP", scapy_name: "IP" },
+    { name: "UDP", scapy_name: "UDP" },
+    { name: "TCP", scapy_name: "TCP" },
+    { name: "DNS", scapy_name: "DNS" },
+    { name: "GRE", scapy_name: "GRE" },
+    { name: "IPv6", scapy_name: "IPv6" },
+    { name: "ICMP", scapy_name: "ICMP" },
+    { name: "Raw", scapy_name: "Raw" },
+    { name: "ARP", scapy_name: "ARP" },
+    { name: "VXLAN", scapy_name: "VXLAN" },
+  ];
+  packet = "";
+  pickedPackets: Array<any> = [];
+  displayPackets: Array<any> = [];
+  res: any;
+  inputs: Array<any> = [];
+  stepper = 0;
+  filter = "";
+  createdHex = "";
+  upperCase = false;
+  spacing = true;
+  successSw = false;
+  alertSw = false;
+  errorSw = false;
+  timeout = 5000;
+
+  get filtered_packets(): Array<any> {
```

Comment:
`filtered_packets()` operates on `packets` array so it can return an array of type Packet